Mixxx 2.1 updates for DJ console 4-Mx

JosepMaJAZ opened the pull request:
2.1 effect racks
2.1 softTakeover fix for setParameter.
xml mapping fixes
sampler mapping fixes
single jog mode hack

JosepMaJAZ committed:
Mixxx 2.1 updates

JosepMaJAZ committed:
Mixxx 2.1 updates

JosepMaJAZ committed:
Mixxx 2.1 updates

JosepMaJAZ committed:
Mixxx 2.1 updates

JosepMaJAZ committed:
Mixxx 2.1 updates

JosepMaJAZ committed:
Mixxx 2.1 updates

Be-ing commented:
Thanks for submitting this. I won't take a detailed look until after 2.1 beta is released, but at first glance I see a lot of legacy code for 2.0. Mixxx 2.0 was already released with a mapping for this controller and this PR will be modifying it for 2.1, so I don't think there's any need to maintain backwards compatibility with 2.0.

Be-ing commented:
I will get to reviewing this soon, but for now I will be taking a break from development and code review for a few days.

daschuer committed:
Merge remote-tracking branch 'upstream/2.1'

sblaisot committed:
Happy New Year 2018!

rryan committed:
Merge pull request #1452 from sblaisot/master-2k18

Happy New Year 2018!

sblaisot committed:
remove NSIS files as we now have completely switched to wix

sblaisot committed:
also remove nsis patch

sblaisot committed:
properly update LICENCE and README.md for 2.2

sblaisot committed:
Remove visual studio version from LICENCE

Be-ing committed:
Merge pull request #1456 from sblaisot/master-readme-update

properly update LICENCE and README.md for 2.2

rryan committed:
Merge branch '2.1'

Be-ing committed:
Merge remote-tracking branch 'upstream/2.1'

Be-ing committed:
Merge pull request #1455 from sblaisot/remove_nsis

Remove NSIS files as we now have completely switched to wix

rryan committed:
Merge branch '2.1'

daschuer committed:
Merge remote-tracking branch 'upstream/2.1'

daschuer committed:
Merge remote-tracking branch 'upstream/2.1'

ferranlala committed:
Decrease Reverb Decay

johanLsp committed:
Fix crash when parsing bad xml controller file

johanLsp committed:
Add error dialog

Be-ing committed:
Merge pull request #1502 from ferranpujolcamins/Decrease-Reverb-Decay

Decrease Reverb Decay

johanLsp committed:
Fix deck state when loading empty track

daschuer committed:
Merge pull request #1507 from johanLsp/lp1746325_play_empty_deck

lp1746325 : Fix empty deck can be played

johanLsp committed:
Revert "Add error dialog"

This reverts commit 70b6c95a6f1e37dfa81d9ca25b4a59dc0d6942cc.

daschuer committed:
Merge pull request #1506 from johanLsp/lp1689014_xmlparser

lp1689014 : Fix crash when parsing bad xml controller file

daschuer committed:
Merge remote-tracking branch 'upstream/2.1'

Commit:
Shift+Filter can control either FX4 or channel gain

Commit:
ShiftFilterFX4 default  value set to true

daschuer committed:
Merge remote-tracking branch 'upstream/2.1'

Be-ing committed:
Merge pull request #1528 from nttmusic/mixtrack3

Numark Mixtrack 3 - Shift+Filter can control either FX4 or channel gain

ywwg committed:
Fix library select knobs that pass through their values directly

Create a new ControlObject type for this kind of unfiltered midi-to-value control.

daschuer committed:
Merge remote-tracking branch 'upstream/2.1'

ywwg committed:
Rename direct control object to Encoder

also update a couple more examples of this use of COs

daschuer committed:
Merge pull request #1546 from ywwg/select-knob-fix

Fix library select knobs that pass through their values directly

daschuer committed:
Merge branch '2.1'

Be-ing commented on res/controllers/Hercules-DJ-Console-4-Mx-scripts.js:
```diff
@@ -93,7 +95,10 @@ Hercules4Mx.userSettings = {
+    'FXbuttonsSetup': 'mixxx21',
+    // This allows to use the left jog wheel in place of the right jog wheel when pressing the right jog wheel.
+    // I implemented this to overcome the failure of the right jog wheel on my controller.
+    'useSingleJogWeelHack': false
```
typo: useSingleJogW**h**eelHack

Be-ing commented on res/controllers/Hercules-DJ-Console-4-Mx-scripts.js:
```diff
@@ -1620,6 +1648,8 @@ Hercules4Mx.getNewDestinationChannel = function(chan) {
 Hercules4Mx.reinitButtonsMap = function() {
     Hercules4Mx.buttonMappings = Hercules4Mx.buttonMappings.slice(Hercules4Mx.buttonMappings.length);
+    //This is setup here because the function is undefined when the map is defined.
+    Hercules4Mx.noActionButtonMap.buttonPressAction=Hercules4Mx.FxActionNoOp;
```
spaces around `=` operator

Be-ing reviewed (changes requested):
Thank you for updating this and sorry for the delay reviewing it. I noted just a few minor nitpicks in the code. Could you remove the legacy code for 2.0? I don't think there's any point keeping it in 2.1.

Be-ing commented:
Also, can you rebase this on the 2.1 branch?

JosepMaJAZ committed:
changes from https://github.com/mixxxdj/mixxx/pull/1419

JosepMaJAZ committed:
Merge branch 'dj_console_v3' of https://github.com/JosepMaJAZ/mixxx into dj_console_v3

JosepMaJAZ committed:
Merge branch 'master' into dj_console_v3

JosepMaJAZ commented:
Glups.... I believe I made a mistake with the rebase... Can you simply get the latest changes and commit them as they should?

Be-ing commented:
I rebased this branch and directly pushed it to 2.1. Thank you @JosepMaJAZ.

Be-ing commented:
Could you checkout the 2.1 branch and confirm that I didn't mess anything up with the rebasing?

JosepMaJAZ commented:
I can confirm that the files have the expected changes.
